### DISCUSSION TITLE
Creating a New Collection

### DISCUSSION THREAD

**Discussion by PersonMeetup** [11ty/eleventy#2212] [Q&A] [answered]
Hey there! Trying to do some weird stuff with 11ty and figured this would be the place to ask for help. Especially because I'm kinda stumped on what to do right now.

I'm currently building myself a website, and in it, I'm wanting to add a portfolio section. Ideally, I want to list the various things I've made and be able to sort them by type without having to make individual files for each project. The idea is like how collections currently work, but with my own custom data.

So far I've tried using JS global data to take care of this, but it's left me with two ways to format the JSON that both give me problems. The first way I could format is by grouping everything initially as so:
```json
{
    "games": [
        {
            "title": "Laser Construction",
            "date": "2017-07-05",
            "url": "https://steamcommunity.com/sharedfiles/filedetails/?id=964675754",
            "platform": "Source",
            "postmortem": ""
        },
        {
            "title": "Detour",
            "date": "2018-09-01",
            "url": "https://steamcommunity.com/sharedfiles/filedetails/?id=1499605287",
            "platform": "Source",
            "postmortem": ""
        }
    ]
}
```
While calling the individual groups works perfectly, this gives me problems when trying to call everything I've made in a similar vein to `collections.all`. From what I've tried there doesn't seem to be a way for Nunjucks to output everything without it either not working or being divided by groups. Another Idea I had was to input the JSON as one big list and compile it into groups at build-time, but `module.exports` seems to always submit the data before it's ready.

As for why I don't simply use 11ty's collections, as far as I know, that would mean creating a new file for every project I want on the portfolio. It could be done, but I would rather maintain one file than maintain multiple. I also feel as though it would cause my tags to become messier with what needs to be filtered and excluded, something I'm hoping to avoid.

If more info on what I've tried so far is needed, let me know!

**Comment by cfjedimaster** [comment]
Are you basically asking - how can you iterate over the top level keys of the data? Ie, get games and any other group?

**Reply by PersonMeetup** [reply]
Yep, that was the problem I was having for the first thing I tried.

**Comment by cfjedimaster** [comment, marked as answer]
Check out this SO post, https://stackoverflow.com/questions/20990516/loop-through-object-properties-nunjucks. I did a quick test and can confirm it works. Specifically, if your data was called `stuff.json`, this gets you access to the top level items:

```

{% for thing, items in stuff %}
thing is: {{ thing }} <br/>
{% endfor %}
```
